torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -550,13 +551,20 @@ struct CUDAFusionFunction : public CompiledFusionFunction {
      // it is possible that this is the first cuda call on this thread
      // so make sure we initialize the Driver API's context
      // cudaFree(0) accomplishes this.
-     cudaFree(0);
-
+     CUcontext pctx = 0;
+     TORCH_CU_CHECK(cuCtxGetCurrent(&pctx));
+     if (!pctx) {
+        std::unique_lock<std::mutex> cudaFreeMutexLock(
+        *(THCCachingAllocator_getCudaFreeMutex()));
+        cudaFree(0);
+        cudaFreeMutexLock.unlock();
```

Comment:
This is unnecessary. It’s going to be unlocked when it leaves the scope which happens just after this line